src/main/thrift/parquet.thrift

```diff
@@ -686,6 +702,9 @@ struct ColumnChunk {
 
   /** Size of ColumnChunk's ColumnIndex, in bytes **/
   7: optional i32 column_index_length
+  
+  /** Column-specific crypto metadata (for non-uniform encryption only) **/
+  8: optional ColumnCryptoMetaData crypto_meta_data
```

Comment:
I guess for backward compatibility the API in parquet-mr should handle the case where crypto_meta_data is null: it is not encrypted. How about dropping 'encrypted' field in ColumnCryptoMetaData, since if crypto_meta_data == null, then the columnChunk is not encrypted (plaintext), and if it isn't null, then the ColumnCryptoMetaData includes the relevant metadata for the decryption. @ggershinsky what do you think about this approach?

Reply:
- backwards compatibility won't be possible for the current readers, they won't be able to read the encrypted parquet files (different magic number, FileCryptoMetaData, encrypted and shifted FileMetaData, etc). 
- Using "crypto_meta_data == null" convention for plaintext columns should be possible. It wont help with backwards compatibility, and makes things a little less formal / explicit. However, on the positive side, it saves some storage space and processing time. Looks like a reasonable thing to do.

Reply:
Indeed, encrypted files can't be read by old readers. But I wanted to refer to this case: non-encrypted files written by "old" writers, where the schema doesn't have crypto_meta_data. That's why I recommended crypto_meta_data == null, hoping that it makes the schema even simpler.

Reply:
These files have a regular ("PAR1") magic number, so a reader wont look for crypto_meta_data. But in any case, switching to the "crypto_meta_data == null" convention seems to be the right thing to do.

Reply:
It's true that old readers won't be able to read the file if `encrypted == true`. However, both `encrypted == false` and `crypto_meta_data == null` mean the same thing (and these files can be read by both old and new readers). For this reason `encrypted` seems to be redundant, but I'm not against keeping it in some form if you prefer to make this information explicit. However, in that case it should be a third value in the "union-as-enum" instead, because it's mutually exclusive to set an encryption and to set encrypted to false.

    union EncryptionMetadata {
      1: EncryptionNone ENCRYPTION_NONE
      2: EncryptionWithFooterKey ENCRYPTION_WITH_FOOTER_KEY
      3: EncryptionWithColumnKey ENCRYPTION_WITH_COLUMN_KEY
    }
    
    struct EncryptionNone {
    }
    
    struct EncryptionWithFooterKey {
    }
    
    struct EncryptionWithColumnKey {
      1: binary column_key_metadata
    }

Reply:
Yep, this is too much overhead to pay for a bit of formality. Lets go with the null convention.